src/methods/extract.jl

```diff
@@ -66,9 +66,17 @@ function extract(x::RasterStackOrArray, data;
     names = NamedTuple{names}(names)
     if Tables.istable(data)
         geomcolnames = GI.geometrycolumns(data)
-        isnothing(geomcolnames) && throw(ArgumentError("No `:geometry` column and `GeoInterface.geometrycolums(::$(typeof(data)))` does not define alternate columns"))
-        geometries = Tables.getcolumn(Tables.columns(data), first(geomcolnames))
-        _extract(T, x, geometries; dims, names, kw...)
+        if isnothing(geomcolnames)
```

Comment:
Hmm is it ever `nothing` ? Or will it fall back to `:geometry` ? 

I think we may need to check `first(geomcolnames) in Tables.columnames(data)`